articles/types/jsdoc.md

```diff
@@ -1,3 +1,179 @@
 # JsDoc support
 
-TODO
+Programs written in JavaScript and TypeScript can be annotated with (JsDoc comments)[http://usejsdoc.org/index.html]
+to give type information and quick-info descriptions of identifiers.
+
+In the following example, the text between the opening `/**` and closing `/*` are considered JsDoc comments.
+
+```js
+/**
+ * @param {"cat" | "dog"} kind  The type of animal to create.
+ * @param name {string}
+ * @param {number} weight The weight, in pounds, of animal.
+ * @param {...string} extraProps Any extra properties to tack on the animal.
+ * @return {{[index: string]: string | number}} Any extra properties to tack on the animal.
+ */
+function createAnimal(kind, name, weight, ...extraProps)
+{
+    // ...
+}
+```
+
+As you can see, we can prefix sections in JsDoc with terms such as `@param` and `@return` to describe the behavior of the `createAnimal` function.
+In JsDoc, terms beginning with `@` have special meanings, and are called *tags*. The text between curly braces are descriptions of *types*,
+which can be used by the developer to check that they are using a variable correctly -- they can go either before or after the construct's *name*.
+In the example above, knowing that `weight` is a number can inform the developer to avoid calling `weight.length`.
+Moreover, if the result of `createAnimal` is assigned to a variable, the return type of createAnimal propogates to the assigned value.
+
+Finally, the text after the parameter names are *descriptions*, which show up as part of *quickinfo* anytime you hover over a construct that is documented
```

Comment:
"text after the parameter names is". I think? Verb agreement in English is surprisingly hard, so maybe ask somebody who natively speaks a language with more verb inflections.

Reply:
Is *quickinfo* in italics because you meant to define it right here?